# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откроем таблицу и изучим общую информацию о данных

**Импортируем библиотеку pandas. Посчитаем данные из csv-файла в датафрейм и сохраним в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [722]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Выведем первые 20 строчек датафрейма `data` на экран.**

In [723]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Выведем основную информацию о датафрейме с помощью метода `info()`.**

In [724]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Выведем количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [725]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [726]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработаем значения в этом столбце: заменим все отрицательные значения положительными с помощью метода `abs()`.**

In [727]:
data['days_employed'] = data['days_employed'].abs()

**Для каждого типа занятости выведем медианное значение трудового стажа `days_employed` в днях.**

In [728]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправим такие значения сложно, поэтому оставим их как есть. Тем более этот столбец не понадобится нам для исследования.

**Выведем перечень уникальных значений столбца `children`.**

In [729]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**В столбце `children` есть два аномальных значения. Удалим строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [730]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Ещё раз выведем перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [731]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Заполним пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [732]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Убедимся, что все пропуски заполнены. Проверим ещё раз, выведем количество пропущенных значений для каждого столбца с помощью двух методов.**

In [733]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [734]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Обработаем неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведем их к нижнему регистру. Проверим остальные столбцы.**

In [735]:
data['education'] = data['education'].str.lower()

**Выведем на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалим их.**

In [736]:
data.duplicated().sum()

71

In [737]:
data = data.drop_duplicates()

### Категоризация данных

**На основании диапазонов, указанных ниже, создадим в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используем собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [738]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [739]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [740]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используем собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучим данные в столбце `purpose` и определим, какие подстроки помогут нам правильно определить категорию.**

In [741]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [742]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуем данные и ответим на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [743]:
#children_debt=data.groupby('children').agg({'debt':['count','sum']})
#children_debt

`Оставлю закомментированными свои первичные действия и добавлю новые`

In [744]:
#children_proportion=children_debt['debt']['sum']/children_debt['debt']['count']
#children_proportion

In [745]:
# Так как в нашей стране с 3 и более детьми считаются многодетными семьями, то лучше объединить эти группы
def make_children_category(children):
    if children == 1:
        return '1 ребенок'
    if children == 2:
        return '2 ребенка'
    if children >= 3:
        return 'Много детей'
    return 'Нет детей'

# добавим столбец children_category в общую таблицу
data['children_category'] = data['children'].apply(make_children_category)
data # проверим таблицу
# подсчитаем значения каждой категории
data['children_category'].value_counts()

Нет детей      14091
1 ребенок       4808
2 ребенка       2052
Много детей      380
Name: children_category, dtype: int64

`Как же много клиентов из категории без детей! Интересно, вовремя ли они погашают свои задолжности?`

In [746]:
# определим количество задолжности клиентов с детьми и без 
children_debt=data.groupby('children_category').agg({'debt':['count','sum']})
children_debt

debt      
                   count   sum
children_category             
1 ребенок           4808   444
2 ребенка           2052   194
Много детей          380    31
Нет детей          14091  1063

In [747]:
# посчитаем долю просрочек по отношению к количеству детей, умножим 100 для лучшего визуального изучения
children_proportion=(children_debt['debt']['sum']/children_debt['debt']['count'])*100
children_proportion

children_category
1 ребенок      9.234609
2 ребенка      9.454191
Много детей    8.157895
Нет детей      7.543822
dtype: float64

`Видим, что количество просрочек у заемщиков без детей составляет 1063, это больше,чем в 1,5 раза больше,чем у всех клиентов с детьми. Но если смотреть их долю в 7%, можно сказать, что эта группа не самая безответственная. Судя по цифрам, семьи с детьми, причем с любым количеством детей, имеют задолжности и, возможно, банк предусматривает данный риск и повышает проценты.`

**Вывод:** Видим, что количество просрочек у заемщиков без детей составляет 1063, это больше,чем в 1,5 раза больше,чем у всех клиентов с детьми. Но если смотреть их долю в 7%, можно сказать, что эта группа не самая безответственная. Судя по цифрам, семьи с детьми, причем с любым количеством детей, имеют задолжности и, возможно, банк предусматривает данный риск и повышает проценты.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [748]:
data # вспомним данные таблицы

children  days_employed  dob_years education  education_id  \
0             1    8437.673028         42    высшее             0   
1             1    4024.803754         36   среднее             1   
2             0    5623.422610         33   среднее             1   
3             3    4124.747207         32   среднее             1   
4             0  340266.072047         53   среднее             1   
...         ...            ...        ...       ...           ...   
21520         1    4529.316663         43   среднее             1   
21521         0  343937.404131         67   среднее             1   
21522         1    2113.346888         38   среднее             1   
21523         3    3112.481705         38   среднее             1   
21524         2    1984.507589         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем                 0      F   сотрудник     0   
2       женат / замужем                 0      M   сотрудник     0   
3       женат / замужем                 0      M   сотрудник     0   
4      гражданский брак                 1      F   пенсионер     0   
...                 ...               ...    ...         ...   ...   
21520  гражданский брак                 1      F   компаньон     0   
21521   женат / замужем                 0      F   пенсионер     0   
21522  гражданский брак                 1      M   сотрудник     1   
21523   женат / замужем                 0      M   сотрудник     1   
21524   женат / замужем                 0      F   сотрудник     0   

       total_income                       purpose total_income_category  \
0            253875                 покупка жилья                     B   
1            112080       приобретение автомобиля                     C   
2            145885                 покупка жилья                     C   
3            267628    дополнительное образование                     B   
4            158616               сыграть свадьбу                     C   
...             ...                           ...                   ...   
21520        224791             операции с жильем                     B   
21521        155999          сделка с автомобилем                     C   
21522         89672                  недвижимость                     C   
21523        244093  на покупку своего автомобиля                     B   
21524         82047         на покупку автомобиля                     C   

               purpose_category children_category  
0      операции с недвижимостью         1 ребенок  
1        операции с автомобилем         1 ребенок  
2      операции с недвижимостью         Нет детей  
3         получение образования       Много детей  
4            проведение свадьбы         Нет детей  
...                         ...               ...  
21520  операции с недвижимостью         1 ребенок  
21521    операции с автомобилем         Нет детей  
21522  операции с недвижимостью         1 ребенок  
21523    операции с автомобилем       Много детей  
21524    операции с автомобилем         2 ребенка  

[21331 rows x 15 columns]

In [749]:

# группируем столбцы о семейном положении и просрочке
family_debt=data.groupby('family_status').agg({'debt':['count','sum']})
family_debt


debt     
                       count  sum
family_status                    
Не женат / не замужем   2796  273
в разводе               1189   84
вдовец / вдова           951   63
гражданский брак        4134  385
женат / замужем        12261  927

In [750]:
# находим долю
family_proportion=family_debt['debt']['sum']/family_debt['debt']['count']
family_proportion

family_status
Не женат / не замужем    0.097639
в разводе                0.070648
вдовец / вдова           0.066246
гражданский брак         0.093130
женат / замужем          0.075606
dtype: float64

In [751]:
# таблицу группируем по столбцу family_status и выводим числовые значения столбца debt
data_pivot_family_status = data.pivot_table(index=['family_status'],
                                        columns='family_status_id', values='debt', aggfunc='sum')
data_pivot_family_status.head(10)

family_status_id           0      1     2     3      4
family_status                                         
Не женат / не замужем    NaN    NaN   NaN   NaN  273.0
в разводе                NaN    NaN   NaN  84.0    NaN
вдовец / вдова           NaN    NaN  63.0   NaN    NaN
гражданский брак         NaN  385.0   NaN   NaN    NaN
женат / замужем        927.0    NaN   NaN   NaN    NaN

In [752]:
data_pivot_family_status = data.pivot_table(index=['family_status'],
                                        columns='family_status_id', values='debt', aggfunc='mean')
data_pivot_family_status

family_status_id              0        1         2         3         4
family_status                                                         
Не женат / не замужем       NaN      NaN       NaN       NaN  0.097639
в разводе                   NaN      NaN       NaN  0.070648       NaN
вдовец / вдова              NaN      NaN  0.066246       NaN       NaN
гражданский брак            NaN  0.09313       NaN       NaN       NaN
женат / замужем        0.075606      NaN       NaN       NaN       NaN

In [753]:
# добавлю агрегированную функцию для проверки и корректности выводов
family_grouped = data.groupby('family_status').agg({'debt':['count','sum', 'mean']})
family_grouped

debt               
                       count  sum      mean
family_status                              
Не женат / не замужем   2796  273  0.097639
в разводе               1189   84  0.070648
вдовец / вдова           951   63  0.066246
гражданский брак        4134  385  0.093130
женат / замужем        12261  927  0.075606

**Вывод:** Самая большая доля просрочек у людей, состоящих в браке(надеятся на лучшее и на супругов) и у клиентов без официальных отношений("ветер в моей голове"). А клиенты, побывавшие в браке, реже допускают просрочки по кредитам.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

напоминание групп
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'.
`


In [765]:
# создадим функцию для группировки данных
# сгруппируем данные по col и debt
# count покажет общее количество наблюдений в данной категории 
# sum просуммирует все единички (т.е. невозвраты), и мы получим количество клиентов, у которых есть просрочка
# сохраним полученные данные в ДатаФрейм
# рассчитаем % невозврата в каждой группе и сохраним результат в столбец debt_proportion
# рассчитаем % невозврата в разрезе групп col
# для этого количество невозврата по данной группе разделим на общее количество невозвратов
def data_group(data,col):
    data_debt = data.groupby(col)['debt'].agg(['count','sum'])
    data_debt['debt_proportion'] = data_debt['sum']/data_debt['count']*100
    data_debt['debt_group']=data_debt['sum']/data['debt'].sum()*100
    return data_debt

In [766]:
data_total_income_debt = data_group(data,'total_income_category')
data_total_income_debt

count   sum  debt_proportion  debt_group
total_income_category                                          
A                         25     2         8.000000    0.115473
B                       5014   354         7.060231   20.438799
C                      15921  1353         8.498210   78.117783
D                        349    21         6.017192    1.212471
E                         22     2         9.090909    0.115473

In [768]:
# добавлю свою сводную таблицу для выводов
# группируем по total_income_category, числовые значения-это debt
data_pivot_total_income = data.pivot_table(index=['total_income_category'],
                                        columns='gender', values='debt', aggfunc='sum')
data_pivot_total_income

gender                     F      M  XNA
total_income_category                   
A                        0.0    2.0  NaN
B                      162.0  192.0  0.0
C                      810.0  543.0  NaN
D                       15.0    6.0  NaN
E                        2.0    0.0  NaN

In [769]:
print(data['total_income'].mean())

165342.74576906848


**Вывод:** Видим из таблиц, что основная группа клиентов банка приходится на категории B(200001–1000000) и C(50001–200000). Интересно то, что с понижением дохода клиенты имеют бОльший процент просрочек. Клиенты категории B имеют 7% невозратов, а заемщики группы C - 8,5%.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [760]:
# посмотрим, как распределились цели кредитов
data['purpose'].value_counts()

свадьба                                   790
на проведение свадьбы                     763
сыграть свадьбу                           760
операции с недвижимостью                  672
покупка коммерческой недвижимости         658
покупка жилья для сдачи                   649
операции с жильем                         647
операции с коммерческой недвижимостью     645
жилье                                     641
покупка жилья                             640
покупка жилья для семьи                   637
недвижимость                              631
строительство собственной недвижимости    628
операции со своей недвижимостью           623
строительство жилой недвижимости          620
покупка своего жилья                      619
строительство недвижимости                619
покупка недвижимости                      616
ремонт жилью                              604
покупка жилой недвижимости                602
на покупку своего автомобиля              504
заняться высшим образованием      

In [774]:
# воспользуемся вышеуказанной функцией и посмотрим на таблицу
data_purpose_debt = data_group(data,'purpose_category')
data_purpose_debt

count  sum  debt_proportion  debt_group
purpose_category                                                 
операции с автомобилем     4279  400         9.347978   23.094688
операции с недвижимостью  10751  780         7.255139   45.034642
получение образования      3988  369         9.252758   21.304850
проведение свадьбы         2313  183         7.911803   10.565820

In [773]:
# добавлю свою сводную таблицу для более сжатых выводов данных
# группируем по purpose_category, числовые значения-это debt, и мне интересен пол
data_pivot_purpose = data.pivot_table(index=['purpose_category'],
                                        columns='gender', values='debt', aggfunc='sum')
data_pivot_purpose

gender                        F      M  XNA
purpose_category                           
операции с автомобилем    232.0  168.0  NaN
операции с недвижимостью  444.0  336.0  0.0
получение образования     206.0  163.0  NaN
проведение свадьбы        107.0   76.0  NaN

**Вывод:** Беря деньги на покупку авто и образование, клиенты банка не спешат оплачивать долги. Самая маленькая просрочка у должников за недвижимость. Видимо, потому что долг и срок слишком большие и хочется побыстрее избавиться от них. 

#### 3.5 Приведем возможные причины появления пропусков в исходных данных.

`Причинами появления пропусков в данных могут быть: сбой в системе программирования банка, нежелание клиентов делиться необходимой информацией, невнимательность оператора банка.`

#### 3.6 Объясним, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

`Для заполнения пропусков значений лучше использовать медианное значение, так как оно является более точным и достоверным источником информации по сравнению со средним значением. Количественные данные могут сильно отличаться друг от друга, и среднее значение в анализе даст неверный вывод`

`Ради интереса и для общих выводов сделаем еще пару небольших выборок`

In [763]:
print(data['dob_years'].mean()) # средний возраст заемщика

43.27795227603019


In [775]:
print(data['total_income'].median()) # медианное значение общего дохода клиентов банка

142594.0


In [764]:
data['debt'].value_counts() # из 20 тыс.клиентов банка 19 тысю не имеют задолженности по возврату кредита,а у 1,7 тыс. заемщиков есть проблемы с возвратом денег.

0    19599
1     1732
Name: debt, dtype: int64

In [786]:
print(data['gender'].value_counts()) # найдем, кто больше берет кредитов, муж или жен

F      14092
M       7238
XNA        1
Name: gender, dtype: int64


### Шаг 4: общий вывод.

В ходе анализа входных данных банка были проверены 4 случая:

- Есть ли зависимость между количеством детей и возвратом кредита в срок?
Зависимость существует, но она незначительная, хотя в ходе анализа было выявлено,что все-таки наличие детей отрицательно сказывается на возврат кредитов.

- Есть ли зависимость между семейным положением и возвратом кредита в срок?
Самыми ненадежными заемщиками банка являются клиенты, не состоящие в браке. Процент просрочек в данной категории составляет 9%. Процент невозврата кредита у клиентов, состоящих в официальном браке составляет 7%. А самыми надежными заемщиками банка являются клиенты, уже побывавшие в браке.

- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?
Конечно, при приеме документов банк в первую очередь обращает внимание на доход клиента, соответственно, зависимость существует. Основная группа клиентов банка приходится на заемщиков с доходом от 5 тыс. до 1 млн. у.е., это категории B и С. Чем ниже доход, тем количество просрочек больше. Клиенты с доходом в 50001–200000(группа С) имеют 8.5% невозвратов, а заемщики с доходом в 200001–1000000(группа B) имеют 7% просрочек. Заемщики группы D с доходом 30001–50000 имеют всего 6% невозратов кредита. Но я думаю,что для более подробного анализа зависимости между уровнем дохода и возвратом кредита не достаточно данных. Мне кажется, нужно учитывать еще срок, на который берут заемщики кредит. Ведь нужно смотреть на ежемесячную сумму. Многие люди, беря кредит, думают, что потянут ежемесячные взносы, а на деле получаются просрочки.

- Как разные цели кредита влияют на его возврат в срок?
Самое большое количество кредитов(10751) клиенты берут на покупку недвижимости, но ответственно выплачивают задолжности. Здесь процент просрочек самый маленький - 7.2%. Так же небольшой процент невозврата кредита в 7.9% у клиентов, пожелавших направить деньги на свадьбу. А самое большое количество просрочек у клиентов, которые берут автокредиты и заем на получение образования. В данных группах по 9% невозврата кредита.

Дополнительные интересности из анализа:
+ средний возраст заемщика составляет 43 года;
+ из 20 тыс.клиентов банка 19 тыс. не имеют задолженности по возврату кредита,а у 1,7 тыс. заемщиков есть проблемы с возвратом денег
+ медианное значение общего дохода среди всех клиентов равняется 142594.0 у.е
+ больше всего берут кредиты женщины. Их количество 14092 против 7238 мужских.

Таким образом, можно сказать: не зря банк очень тщательно рассматривает документы и требует заполнять все пункты по вышеуказанным вопросам. Ведь все анализируемые данные влияют на своевременный возврат денег банку.